_drafts/2021-02-08-custom_markdown_for_github_pages.md

```diff
@@ -0,0 +1,176 @@
+---
+layout: post
+title: "Making Custom Markdown for Github Pages"
+description: >
+  I love Markdown, I take all my notes in it and write my blog in it. But
+  sometimes you want to create new syntax; read on to find out how!
+image: /files/jekyll/stansaab_censor_and_operator.jpg 
+image_alt: >
+  A black and white photo of a man in a short-sleeve collared shirt uses an
+  osciliscope to adjust a circuit board on a Stansaab 'Censor' computer at
+  Standard Radio's factory in Ulvsunda, Stockholm.
+categories:
+  - software-development
+---
+
+{% capture file_dir %}/files/jekyll/{% endcapture %}
+
+[Markdown] is a great way to write; simple enough to be read as text, with all
+the power of HTML. I have been using it for my own notes for a decade and this
+site is written in Markdown using [Jekyll].
+
+Markdown provides a lot of syntax to simplify HTML, like `**BOLD**` to create
+`<strong>BOLD</strong>` text, or `> Quote` to create
+`<blockquote>Quote</blockquote>`. Recently, for my [MiniFate] project, I
+wanted to add a few custom `<span>` elements to highlight specific pieces of
+the text. I could have fallen back to writing it out in HTML each time, but
+doing so felt clunky when compared to how smooth writing Markdown normally is.
+
+[Markdown]: https://en.wikipedia.org/wiki/Markdown
+[Jekyll]: https://en.wikipedia.org/wiki/Jekyll_(software)
+[MiniFate]: https://github.com/MiniFate/MiniFate
+
+I created a way to define my own syntax based on [Anatol Broder's][ab]
+[Compress] and [Sylvain Durand's][sd] post on [_Improving typography on
+Jekyll_][it]. It uses [Liquid] to rewrite the web page **after** it has been
+compiled, allowing complete control on formatting and allowing you to define
+custom Markdown syntax. Since it uses only the default tools built in to
+Jekyll, it works natively on [Github Pages]! Here is how it works:
+
+[ab]: https://bro.doktorbro.net/
+[Compress]: https://jch.penibelst.de/
+[sd]: https://sylvaindurand.org/
+[it]: https://sylvaindurand.org/improving-typography-on-jekyll/
+[Liquid]: https://shopify.github.io/liquid/
+[Github Pages]: https://pages.github.com/
+
+## Layouts
+
+The [order of interpretation][ooi] for Jekyll runs Liquid _before_ compiling
+Markdown to HTML, which means you can't modify the HTML with Liquid. But
+pushing content to a layout happens _after_ the HTML is compiled, and layouts
+can contain their own Liquid code. This allows us to run Liquid on the full
+HTML of the page!
```

Comment:
You can't run Liquid on the HTML, but luckily you can run liquid on the HTML! I think this paragraph could use one more sentence in the middle to clarify